controllers/base.go

```diff
@@ -163,6 +165,57 @@ func (bmrc *baseMachineReconcileContext) releaseHardware() error {
 		return fmt.Errorf("patching Hardware object: %w", err)
 	}
 
+	if err := bmrc.powerOffHardware(hardware); err != nil {
+		return fmt.Errorf("failed to power off hardware: %w", err)
+	}
+
+	return nil
+}
+
+// powerOffHardware uses hardware BMCRef to create a job to power off the machine.
+func (bmrc *baseMachineReconcileContext) powerOffHardware(hardware *tinkv1.Hardware) error {
+	if hardware.Spec.BMCRef == nil {
+		bmrc.log.Info("Skipping hardware power off", "BMCRef", hardware.Spec.BMCRef, "Hardware", hardware.Name)
+
+		return nil
+	}
+
+	return bmrc.createPowerOffJob(hardware)
+}
+
+// createPowerOffJob creates a BMCJob object with the required tasks for hardware power off.
+func (bmrc *baseMachineReconcileContext) createPowerOffJob(hardware *tinkv1.Hardware) error {
+	powerOffAction := rufiov1.HardPowerOff
+	bmcJob := &rufiov1.BMCJob{
```

Comment:
Can we deduplicate some of this construction and perhaps have a separate function set or builder.

```
func (bmrc *baseMachinereconcileContext) createPowerOffJob(hardware ...) error {
  job := newBMCJobBuilder(bmrc.tinkerbellMachine, hardware)
  job.WithTask(rufiov1.Task{
    PowerAction: *powerOffAction)
  })

  if err := bmrc.client.Create(bmrc.ctx, job.Build()); err != nil {
    ...
  }
}
```

This lets us re-use that construction logic elsewhere in the program like in `ensureHardware`.